Python/Product/Analysis/LanguageServer/Server.Completion.cs

```diff
@@ -25,26 +25,27 @@ namespace Microsoft.PythonTools.Analysis.LanguageServer {
     public sealed partial class Server {
         public override Task<CompletionList> Completion(CompletionParams @params) => Completion(@params, CancellationToken.None);
 
-        internal Task<CompletionList> Completion(CompletionParams @params, CancellationToken cancellationToken) {
+        internal async Task<CompletionList> Completion(CompletionParams @params, CancellationToken cancellationToken) {
             var uri = @params.textDocument.uri;
  
             ProjectFiles.GetAnalysis(@params.textDocument, @params.position, @params._version, out var entry, out var tree);
             TraceMessage($"Completions in {uri} at {@params.position}");
 
             tree = GetParseTree(entry, uri, cancellationToken, out var version) ?? tree;
-            var analysis = entry?.Analysis;
+            var analysis = entry != null ? await entry.GetAnalysisAsync(50, cancellationToken) : null;
```

Comment:
I actually tried this approach a while ago. I used 100 ms I think. But then resolved problem differently. 
The analysis should not be clearing old stuff until new is available. I.e. if class declares variables they should remain until analyzer figures out new set.

Reply:
Well, there is definitely a moment when there are no variables, cleaning doesn't happen as atomic operation. And there is still another concurrency issue when `entry?.Analysis` is accessed before parsing is actually started.